src/Aspire.Hosting/DistributedApplicationBuilder.cs

```diff
@@ -119,6 +124,11 @@ public DistributedApplicationBuilder(DistributedApplicationOptions options)
         LogBuilderConstructed(this);
     }
 
+    private void MapTransportOptionsFromCustomKeys(TransportOptions options)
+    {
+        options.AllowUnsecureTransport = Configuration.GetBool(KnownEnvironmentVariables.AllowUnsecuredTransport, false);
```

Comment:
Only set this if the configuration is present. Allows someone to configure `AllowUnsecureTransport` directly without it always being overriden.

e.g.
https://github.com/dotnet/aspire/blob/23104ee048dcf99bc1bdcb5ff8ad01ee85921d3b/src/Aspire.Dashboard/Configuration/PostConfigureDashboardOptions.cs#L33-L34